code/game/gamemodes/objective.dm

```diff
@@ -17,6 +17,15 @@ GLOBAL_LIST_EMPTY(objectives)
 	if(text)
 		explanation_text = text
 
+/datum/objective/Destroy(force, ...)
+	GLOB.objectives -= src
+	if(owner)
+		for(var/datum/antagonist/A in owner.antag_datums)
```

Comment:
Aren't objectives tied to the mind? At least crew objectives are, so that logic wouldn't work for them.

Reply:
We can make `/datum/objective/crew/Destroy` which takes care of mind's `crew_objectives` list.